addons/web/static/tests/views/fields/badge_selection_with_filter_field.test.js

```diff
@@ -0,0 +1,201 @@
+            mood: "happy",
+            color: "white",
+        },
+    ];
+}
+
+defineModels([Partner]);
+
+test("badge selection field with filter, empty list", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 1,
+        arch: /* xml */ `
+            <form>
+                <field name="color" widget="selection_badge_with_filter"
+                       context="{'allowed_selection': []}" />
+            </form>
+        `,
+    });
+
+    expect(".o_selection_badge").not.toBeVisible();
+});
+
+test("badge selection field with filter, single choice", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 1,
+        arch: /* xml */ `
+            <form>
+                <field name="color" widget="selection_badge_with_filter"
+                       context="{'allowed_selection': ['grey']}" />
+            </form>
+        `,
+    });
+
+    expect(".o_selection_badge").toHaveCount(1);
+    expect(".o_selection_badge[value='\"white\"']").not.toBeVisible();
+    expect(".o_selection_badge[value='\"grey\"']").toBeVisible();
+    expect(".o_selection_badge[value='\"black\"']").not.toBeVisible();
+});
+
+test("badge selection field with filter, all choices", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 1,
+        arch: /* xml */ `
+            <form>
+                <field name="color" widget="selection_badge_with_filter"
+                       context="{'allowed_selection': ['white', 'grey', 'black']}" />
+            </form>
+        `,
+    });
+
+    expect(".o_selection_badge").toHaveCount(3);
+    expect(".o_selection_badge[value='\"white\"']").toBeVisible();
+    expect(".o_selection_badge[value='\"grey\"']").toBeVisible();
+    expect(".o_selection_badge[value='\"black\"']").toBeVisible();
+});
+
+test("badge selection field with filter, synchronize with other field", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 1,
+        arch: /* xml */ `
+            <form>
+                <group>
+                    <field name="is_raining_outside" />
+                    <field name="mood" widget="selection_badge_with_filter"
+                           context="{'allowed_selection':
+                                ['happy']
+                                + (['sad'] if is_raining_outside else [])
+                           }" />
+                </group>
+            </form>
+        `,
+    });
+    // not raining outside => sad should be invisible
+    expect("[name='is_raining_outside'] input").not.toBeChecked();
+    expect("div[name='mood'] .o_selection_badge").toHaveCount(1);
+    expect(".o_selection_badge[value='\"happy\"']").toBeVisible();
+    expect(".o_selection_badge[value='\"sad\"']").not.toBeVisible();
+
+    await click("[name='is_raining_outside'] input");
+    await animationFrame();
```

Comment:
is this animation frame necessary?

Reply:
I will check when the compute work (see my general comment).